src/restarts/restarts_manager.hpp

```diff
@@ -100,8 +128,18 @@ RestartsManager<Writer>::addRestartDict(initializer::PHAREDict const& params)
         throw std::runtime_error("Restarts invalid, properties already set");
 
     restarts_properties_ = std::make_unique<RestartsProperties>();
-    restarts_properties_->writeTimestamps
-        = params["write_timestamps"].template to<std::vector<double>>();
+
+    if (params.contains("write_timestamps"))
+        restarts_properties_->writeTimestamps
+            = params["write_timestamps"].template to<std::vector<double>>();
+
+    if (params.contains("elapsed_timestamps"))
+    {
+        restarts_properties_->elapsedTimestamps
+            = params["elapsed_timestamps"].template to<std::vector<double>>();
+        for (auto& time : restarts_properties_->elapsedTimestamps)
+            time += start_time_; // expected for comparison later
```

Comment:
:small_red_triangle: :small_red_triangle: :small_red_triangle: :small_red_triangle:
hmmm just a question : when the program runs in parallel, don't you expect to have issues ? The clock time has no reason to be the same on all processes. I think we need to sync that somehow.
:small_red_triangle: :small_red_triangle: :small_red_triangle: :small_red_triangle: